File: src/test/java/com/weeth/config/TestContainersTest.java
package com.weeth.config;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.context.annotation.Import;
import org.testcontainers.containers.MySQLContainer;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

@DataJpaTest
@Import(TestContainersConfig.class)
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class TestContainersTest {

    @Autowired
    private MySQLContainer mysqlContainer;

    @Test
    void 설정파일로_주입된_컨테이너_정상_동작_테스트() {
        assertThat(mysqlContainer).isNotNull();
        assertThat(mysqlContainer.isRunning()).isTrue();
        System.out.println("Container JDBC URL: " + mysqlContainer.getJdbcUrl());
    }
}

File: src/test/kotlin/com/weeth/config/TestContainersTest.kt
package com.weeth.config

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.booleans.shouldBeTrue
import io.kotest.matchers.nulls.shouldNotBeNull
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.context.annotation.Import
import org.testcontainers.containers.MySQLContainer

@DataJpaTest
@Import(TestContainersConfig::class)
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class TestContainersTest(
    private val mysqlContainer: MySQLContainer<*>,
) : StringSpec({

        "설정파일로 주입된 컨테이너 정상 동작 테스트" {
            mysqlContainer.shouldNotBeNull()
            mysqlContainer.isRunning.shouldBeTrue()
        }
    })
